### PR TITLE
[WindTunnel] Scenario documentation

### PR DESCRIPTION
**Pull request by IvanPombo**
This introduces a README.md file specific for the WindTunnel scenario with its main functionalities and post-processing capabilities.

### CODE REVIEW

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Wind Tunnel Scenario
+
+This scenario models the aerodynamics of an object inside a virtual wind tunnel for a given airflow velocity. Air is injected on a side wall of the wind tunnel, the flow changes according to the structure of the object and leaves through an outlet on the other side. The system is modelled with the steady-state equations for incompressible flow and the $k-\epsilon$ turbulence models.
+
+To initialize the scenario, the user can define the following parameters:
+- Dimensions of the wind tunnel in meters, e.g. `{"x": [-5, 15], "y": [-5, 5], "z": [0, 8]}`.
+- Airflow velocity in m/s, e.g. `[30, 0, 0]`. Notice that the airflow is injected from the negative x-direction.
+
+Now, the user is ready to simulate the steady state. Here, the user chooses the object to be inserted inside the wind tunnel. This object is defined with a geometry file in STL or OBJ format (currently tested). Notice, that appropriate meshing will be done during the simulation.
```
[the remark this answers is not in the document]
Comment (sarmento) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:9]: Notice that (no comma) the required meshing step will *automatically* be made BEFORE starting

(maybe you want to give mode details about this. what parameters are going to be default for this meshing operation. Meshing is so important that we need a bit more detail here).

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+## Output and Post-Processing:
+
+The `WindTunnel` scenario allows users to extract metrics about the object under the established flow velocity.
+The simulation outputs of the wind tunnel are:
+- `pressure_field`: Pressure field of the airflow over the object.
+- `streamlines`: Streamlines of the airflow over the domain and interacting with the object.
+- `flow_slice`: Slice of the flow that represents the airflow over the domain and interacting with the object.
+- `force_coefficients`: Force coefficients that represent the forces acting on the object. These are the drag and lift coefficients.
+
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:47]: I think you need to explain a bit more what are the data structures involved here. Are these numpy classes?

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+### Default vs Full Post-processing Tools
+
+Since at times the output files can be large, the user can choose to get only some of the default outputs that are post-processed
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:50]: It's the first time you are talking about files. What files? Maybe we need to have an explanation / reminder of how data flows before we go to this section.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                        save_path="default_flow_slice.png")
+```
+
+<img src="../../../../resources/media/openfoam/default_flow_slice.png" width="400" height="300" />
+
+### Example for general Post-processing
+
+```python
+# Get all the WindTunnel simulation files
+output = task.get_output(full=True)
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:110]: I am not conforable with the parameter "full". It is too vague...

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+# Render
+pressure_field.render_frame(save_path="pressure_field.png")
+```
+
+<img src="../../../../resources/media/openfoam/pressure_field.png" width="400" height="300" />
+
+```python
+# Get the streamlines mesh
+streamlines = output.get_streamlines(max_time=200,
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:123]: Here is a general comment, for future, and I think you need to address this with @hpenedones 
I think if you hang everything in the output object this sort of objects becomes overly complex, and they also will become difficult to document. And, worse, their logic will vary imensly from simulator to simulator.

For now, let's not change anything BUT I think we should consider moving to a world where the output objects are basically interfaces between the result data and the users, and they basically know about where data lives. They don't have to do any specialized processing: they are just interfaces to data, they know where data is located, and they are able to move data around.

Once you have one of those output object, then you can basically pass it to a PostProcessing object (which can transform, visualize, combine, etc the data). For example, here I would prefer to have something like>

```
output = task.get_output()

streams = StreamLinesProcessor(output)
# At this point StreamLinesProcessor doesn't care about where data lives.
# but the output object knows where data lives in the server. So, even visualization
# methods can be run on the server.
streams.plot("name.png", other_params=...)
```
This pattern can be repeated for every scenario.....

@hpenedones @fabiocruz @cristianacarpinteiro @david451lima : wdyt? Do you see where I am trying to go?

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
+# Wind Tunnel Scenario
+
+This scenario models the aerodynamics of an object inside a virtual wind tunnel for a given airflow velocity. Air is injected on a side wall of the wind tunnel, the flow changes according to the structure of the object and leaves through an outlet on the other side. The system is modelled with the steady-state equations for incompressible flow and the $k-\epsilon$ turbulence models.
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:3]: Split the middle sentence into two, e.g. "Air is injected on a side wall of the wind tunnel and leaves through an outlet on the opposite side. The air flow within the tunnel is modified according to the structure of the object."

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabiocruz) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:9]: I second @sarmento's comments here.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+Now, the user is ready to simulate the steady state. Here, the user chooses the object to be inserted inside the wind tunnel. This object is defined with a geometry file in STL or OBJ format (currently tested). Notice, that appropriate meshing will be done during the simulation.
+
+The user can further define the following simulation parameters:
+- simulator: The simulator to be used for the simulation. Currently, only `OpenFOAM` is supported.
+- num_iterations: Number of iterations for the steady-state simulation.
+- resolution: Resolution of the meshing. The higher the resolution, the finer the meshing.
+
+Moreover, the hardware and interaction are configured with the usual general parameters - `machine_group`, `run_async`, `n_cores`.
+Launching a simulation returns a task object, which can be used to verify the status of the simulation, get the simulation outputs and access instantly post-processing tools. See more in [Tasks](inductiva/tasks/README.md).
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:17]: Remove "instantly".

Comment (fabiocruz) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:17]: Also, link to the right section on the main README file.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    flow_velocity=[30, 0, 0],
+    dimensions={"x": [-5, 15], "y": [-5, 5], "z": [0, 8]})
+
+# Run a simulation
+task = scenario.simulate(
+    object_path="f1.obj",
+    num_iterations=1000, resolution=0.5,
+    run_async=True, n_cores=4)
+
+# Get the simulation output on your local machine.
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:35]: "Download the simulation output to your local machine."

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+## Output and Post-Processing:
+
+The `WindTunnel` scenario allows users to extract metrics about the object under the established flow velocity.
+The simulation outputs of the wind tunnel are:
+- `pressure_field`: Pressure field of the airflow over the object.
+- `streamlines`: Streamlines of the airflow over the domain and interacting with the object.
+- `flow_slice`: Slice of the flow that represents the airflow over the domain and interacting with the object.
+- `force_coefficients`: Force coefficients that represent the forces acting on the object. These are the drag and lift coefficients.
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:46]: You are listing these as if they were variables. How can the user access these variables? I think you should either remove the code formatting and explain in plain english only what the outputs are, or introduce the output class before you refer to variables.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+### Default vs Full Post-processing Tools
+
+Since at times the output files can be large, the user can choose to get only some of the default outputs that are post-processed
+on run-time by just doing `task.get_output()` (e.g., pressure field, streamlines and flow_slice). Otherwise, the user can choose to get all the outputs with `task.get_output(full=True)` and post-process as he wishes locally.
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:51]: "on run-time" is not accurate, since they are produced after the simulation. Rephrase to "remotely"?

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+- `streamlines`: Streamlines of the airflow over the domain and interacting with the object.
+- `flow_slice`: Slice of the flow that represents the airflow over the domain and interacting with the object.
+- `force_coefficients`: Force coefficients that represent the forces acting on the object. These are the drag and lift coefficients.
+
+### Default vs Full Post-processing Tools
+
+Since at times the output files can be large, the user can choose to get only some of the default outputs that are post-processed
+on run-time by just doing `task.get_output()` (e.g., pressure field, streamlines and flow_slice). Otherwise, the user can choose to get all the outputs with `task.get_output(full=True)` and post-process as he wishes locally. 
+
+In any case, to obtain the metrics above the user can use the following post-processing tools:
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:53]: Again, how do I access these functions? Are they methods of a class, or are they stored in some submodule of the package? Clarify that the user can access these from `output`.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+- `get_output_mesh`: Returns the mesh of the airflow over the entire domain and data on the object.
+- `get_object_pressure_field`: Returns the pressure field of the airflow over the object.
+- `get_streamlines`: Input parameters - `max_time`, `n_points`, `initial_step_length`, `source_radius`, `source_center`. Returns a mesh of the streamlines with pressure and velocity components of airflow over the domain.
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:56]: If you mention the input parameters, you need to explain what each of them is.

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 5 above)
Comment (fabiocruz) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:123]: This is a good idea @sarmento. However, the way streamlines (or other data) is defined in data files varies from simulator to simulator (e.g. they may be saved in different file formats, or defined differently within a common file format), so we'd still have to code a streamlines processor for each scenario.

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+                                     initial_step_length=0.1,
+                                     source_radius=0.5,
+                                     source_center=[-3, 0, 1])
+
+# Render the streamlines
+streamlines.render_frame(physical_field="pressure",
+                         flow_cmap="viridis",
+                         view="isometric",
+                         streamline_radius=0.1
+                         save_path="streamlines.png")
```
Comment (fabiocruz) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:134]: Aren't most of these default? Omit them, if so.

**Review by fabiocruz — COMMENTED** [review]
Cool to see this @IvanPombo. I think there is room for improvement, but this is in good overall shape already.

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 2 above)
Comment (IvanPombo) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:47]: I will add them in the get functions.

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 13 above)
Comment (IvanPombo) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:134]: They are, but in this way I give them a preview of the possibilities without having to explain everything at this point.

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 12 above)
Comment (IvanPombo) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:56]: I feel that here that will be to heavy and the argument names are very explicit for those interested in doing post-processing. I think in the future, with the read the docs, that will be simpler to do.

**Comment by luispcunha** [comment]
I think that it's great that we are moving scenario documentation to separate files, but afterwards we need to make sure we have them listed and linked in the main file, and we probably should have a preview (prob visual) of some of the scenarios there in order to have a more appealing main page. This is not specifically related to this PR, but just raising the concern. Wdyt?

**Comment by IvanPombo** [comment]
> I think that it's great that we are moving scenario documentation to separate files, but afterwards we need to make sure we have them listed and linked in the main file, and we probably should have a preview (prob visual) of some of the scenarios there in order to have a more appealing main page. This is not specifically related to this PR, but just raising the concern. Wdyt?

Completely agree here. Probably, this files should live in a different folder in the first inductiva and not in each scenario. I am still not sure how the page should look, i.e., do we keep the code?

**Comment by sarmento** [comment]
> > I think that it's great that we are moving scenario documentation to separate files, but afterwards we need to make sure we have them listed and linked in the main file, and we probably should have a preview (prob visual) of some of the scenarios there in order to have a more appealing main page. This is not specifically related to this PR, but just raising the concern. Wdyt?
> 
> Completely agree here. Probably, this files should live in a different folder in the first inductiva and not in each scenario. I am still not sure how the page should look, i.e., do we keep the code?

I think we should start organizing the documentation in such a way it reflects the organization of the code (i.e. inductiva.[fluids|molecules|coastal, etc]). I am having some trouble "processing" the long top README page we have at github. Ideally the top page would only have information about install, one example or two examples and then a few images. The details of each scenario / simulator would be found inside each of the "areas" we have (fluids, molecules, etc..). We could have a special section for Tasks and other api related features, but these are more "advanced" and the user does not have to see that in the top github page. 

Can we already try to move some of the docs to "separate folders" and link it from the top github readme

WDYT @hpenedones ?

**Comment by IvanPombo** [comment]
@everyone Can anyone check this PR?

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 14]
```diff
@@ -0,0 +1,155 @@
+# Wind Tunnel Scenario
+
+This scenario models the aerodynamics of an object inside a virtual wind tunnel for a given airflow velocity. Air is injected on a side wall of the wind tunnel and leaves through an outlet on the opposite side. The air flow within the tunnel is modified according to the structure of the object. The system is modelled with the steady-state equations for incompressible flow and the $k-\epsilon$ turbulence models. The simulation is currently performed with the OpenFOAM simulator.
```
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:3]: Air is injected through one side of the wind tunnel and exits through an outlet on the opposite side.

File: docs/index.md [hunk 15]
```diff
@@ -1 +1,70 @@
+# Sample index file
```
Comment (sarmento) [inline docs/index.md:1]: Why do we have this file?

**Review by sarmento — COMMENTED** [review]
LGTM!

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+
+To initialize the scenario, the user can define the following parameters:
+- Dimensions of the wind tunnel in meters, e.g. `{"x": [-5, 15], "y": [-5, 5], "z": [0, 8]}`.
+- Airflow velocity in m/s, e.g. `[30, 0, 0]`. Notice that the airflow is injected from the negative x-direction.
+
+Now, the user is ready to simulate the steady state. Here, the user chooses the object to be inserted inside the wind tunnel. This object is defined with a geometry file in STL or OBJ format. Notice that, the required meshing step will automatically be made before starting the simulation. The meshing is done with the snappyHexMesh tool of OpenFOAM.
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:9]: maybe you want to point to some docs about snappyHexMesh

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+
+The user is performing steady-state simulations of a vehicle inside the wind tunnel with air flowing. The simulation uses an iterative algorithm that converges to the steady-state of the flow. 
+The simulation parameters available for the user to configure are:
+- num_iterations: Set the maximum number of iterations for the algorithm to converge.
+- resolution: Controls the resolution of the meshing that is done prior to the simulation. The higher the resolution, the finer the meshing. Possibilities: "high", "medium", "low".
+
+Moreover, the hardware and interaction are configured with the usual general parameters - `machine_group`, `run_async`, `n_cores`.
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:16]: n_cores needs to be revised. Please bring this up to @hpenedones

File: inductiva/fluids/scenarios/wind_tunnel/README.md [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+# Download the simulation output to your local machine.
+output = task.get_output()
+```
+
+This last step is essential to download the simulation files into your local machine 
+and apply the post-processing and visualizations to understand the aerodynamics of the object.
+
+## Output and Post-Processing
+
+The simulation output is comprised of several files spread throughout several directories. These files contain data about the airflow over the object, from which valuable metrics can be extracted to understand the aerodynamics of the object. To obtain these metrics a few post-processing steps are required to be done.
```
Comment (sarmento) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:44]: several .... several....

**Review by sarmento — APPROVED** [review]

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 16 above)
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:9]: remove the comma in "Notice that the..."

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 17 above)
Comment (cristianacarpinteiro) [inline inductiva/fluids/scenarios/wind_tunnel/README.md:16]: what do you mean by interaction?

**Review by cristianacarpinteiro — APPROVED** [review]

File: inductiva/fluids/scenarios/wind_tunnel/README.md (hunk 18 above)
Comment (hpenedones) [reply inductiva/fluids/scenarios/wind_tunnel/README.md:44]: Yeah and try to keep it simple. Perhaps this paragraph can be replaced by just this:

"The output will contain several sub-directories with files that can be post-processed to extract relevant metrics, such as:"
